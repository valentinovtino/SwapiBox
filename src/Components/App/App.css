@font-face {
  font-family: SWheader;
  src: url(../../fonts/Starjout.ttf);
}

@font-face {
  font-family: SWbutton;
  src: url(../../fonts/bad-mofo.regular.ttf);
}

.btn-container {
  /* border: 1px solid black; */
  width: 50%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around
}

.button {
border: none;
width: 100px;
height: 100px;
text-align: center;
background: transparent;
color: white;
font-size: xx-large;
font-family: 'SWbutton';
letter-spacing: 5px;
text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

/* .button:hover {
  color: yellow;
  text-shadow: -1px 0 yellow, 0 1px yellow, 1px 0 yellow, 0 -1px yellow;
} */


.title {
  /* color: yellow; */
  text-shadow: -1px 0 yellow, 0 1px yellow, 1px 0 yellow, 0 -1px yellow;
  text-align: center;
  font-family: 'SWheader';
  font-size: 50px;
  margin: auto;
}


