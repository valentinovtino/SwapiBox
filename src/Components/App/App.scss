.btn-container {
  border: 1px solid black;
  width: 50%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around
}

.button{
border: none;
width: 100px;
height: 100px;
text-align: center;
background: transparent;
color: white;
font-size: xx-large
text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

.button: hover {
  color: yellow;
  text-shadow: -1px 0 yellow, 0 1px yellow, 1px 0 yellow, 0 -1px yellow;
  width: 100px;
height: 100px;
text-align: center;
font-size: xx-large;
}

.title {
  color: yellow;
  text-shadow: -1px 0 yellow, 0 1px yellow, 1px 0 yellow, 0 -1px yellow;
}


