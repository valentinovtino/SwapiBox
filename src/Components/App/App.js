import React, { Component } from 'react';
import './App.css';
import api from '../../api/api.js'
import FilmCrawl from '../FilmCrawl/FilmCrawl.js';
import { CardContainer } from '../CardContainer/CardContainer.js';
import { getHomeWorld, getSpecies, getPlanetDetails, getVehicleDetails } from '../../api/helper.js';
import { fetchApi } from '../../api/api.js';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      film: [],
      cardType: '',
      people: [],
      vehicles: [],
      planets: [],
      cleanedCardData: [],
      initialFetchedObj: [],
      favorite: []
    }
  }

  handleFavClick = () => {
      console.log('wooo')
  }

  handleClick = (event) => { 
    this.setState({ cardType: event.target.value }, () => {this.initialFetchCall()})
  }

  initialFetchCall = async () => {
    const categoryData = await fetchApi(this.state.cardType)

    this.checkCardType(categoryData) 
  }

  getpeopleData = async (categoryData) => {
    const homeWorldObj = await getHomeWorld(categoryData)
    const speciesObj = await getSpecies(categoryData)
    const cleanedCardData = this.combineObj(homeWorldObj, speciesObj)
    
    this.setState({ people: cleanedCardData })
  }

  getVehicleData = async (categoryData) => {
    const vehicleObj = await getVehicleDetails(categoryData)

    this.setState({ vehicles: vehicleObj })
  }
  
  getPlanetData = async (categoryData) => {
    const planetObj = await getPlanetDetails(categoryData)

    this.setState({ planets: planetObj })
  }

  checkCardType = async (categoryData) => {
    if (this.state.cardType === 'people') {
      this.getpeopleData(categoryData);
    }
    else if (this.state.cardType === 'planets') {
      this.getPlanetData(categoryData);
    }
    else if (this.state.cardType === 'vehicles') {
      this.getVehicleData(categoryData)
    }

    localStorage.setItem([this.props.cardType], JSON.stringify(this.state[this.props.cardType]))
  }

  combineObj = (homeWorldObj, speciesObj) => {
    let combo;
    const combinedObj = homeWorldObj.reduce((acc, item, i) => {
      combo = Object.assign({}, speciesObj[i])
      acc.push({ ...item, ...combo })
      return acc
    }, [])
    return combinedObj
  }

  checkForLocalStorage = () => {
    !localStorage[this.props.cardType] ? this.initialFetchCall() : this.getFromLocalStorage()
  }

  getFromLocalStorage = () => {
    const retrieveData = localStorage.getItem(this.props.cardType)
    const parsedData = JSON.parse(retrieveData)

    if (!this.state[this.props.cardType]) {
      this.setState({ [this.props.cardType]: parsedData })
    }

  }

  render() {
    return (
      <div className="App">
        <FilmCrawl />
        <button onClick={this.handleClick} value='people' className='people-btn'>People</button>
        <button onClick={this.handleClick} value='planets' className='button'>Planets</button>
        <button onClick={this.handleClick} value='vehicles' className='button'>Vehicle</button>
        <button>Favorites</button>

        <div>
          {this.state.cardType && <CardContainer handleFavClick={() => this.handleFavClick()} allState={this.state} />}
        </div>

      </div>
    );
  }
}

export default App;
