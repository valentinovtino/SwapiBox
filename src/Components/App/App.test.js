import React from 'react';
import renderer from 'react-test-renderer';
import App from './App.js';
import { shallow } from 'enzyme';

describe('App', () => {
    let app;
    

    beforeEach( async () => {
        app = shallow(<App />, {disableLifecycleMethods: true})

    })

    it.skip('should match the snapshot', () => {
        const appSnapShot = renderer.create(<App />).JSON;

        expect(appSnapShot).toMatchSnapshot();
    })
    it('should have default states', async () => {
        // console.log(wrapper)
       const wrapper = await shallow(<App />)
        expect(wrapper.state().film).toEqual([])
        expect(wrapper.state().people).toEqual([])
        expect(wrapper.state().vehicles).toEqual([])
        expect(wrapper.state().planets).toEqual([])
        expect(wrapper.state().cleanedCardData).toEqual([])

    })

    it('should setState', () => {

    })

    it('should call fetchApi', () => {

    })

    it('should call checkCardType with the correct params', () => {

    })
    
    it('should update vehicle ste when getVehicleData is called', () => {

    })

    it('conbineObj should return correct Obj')

})