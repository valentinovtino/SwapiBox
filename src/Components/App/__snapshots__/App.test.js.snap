// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FilmCrawl />,
        <button
          className="button"
          onClick={[Function]}
          value="people"
>
          People
</button>,
        <button
          className="button"
          onClick={[Function]}
          value="planets"
>
          Planets
</button>,
        <button
          className="button"
          onClick={[Function]}
          value="vehicles"
>
          Vehicle
</button>,
        <button>
          Favorites
</button>,
        <div />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "button",
          "onClick": [Function],
          "value": "people",
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "button",
          "onClick": [Function],
          "value": "planets",
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicle",
          "className": "button",
          "onClick": [Function],
          "value": "vehicles",
        },
        "ref": null,
        "rendered": "Vehicle",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FilmCrawl />,
          <button
            className="button"
            onClick={[Function]}
            value="people"
>
            People
</button>,
          <button
            className="button"
            onClick={[Function]}
            value="planets"
>
            Planets
</button>,
          <button
            className="button"
            onClick={[Function]}
            value="vehicles"
>
            Vehicle
</button>,
          <button>
            Favorites
</button>,
          <div />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "button",
            "onClick": [Function],
            "value": "people",
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "button",
            "onClick": [Function],
            "value": "planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicle",
            "className": "button",
            "onClick": [Function],
            "value": "vehicles",
          },
          "ref": null,
          "rendered": "Vehicle",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
